src/ts/components/core/menu/MenuSub.tsx

```diff
@@ -0,0 +1,65 @@
+import { Box, Menu as MantineMenu } from "@mantine/core";
+import { __PopoverProps } from "props/popover";
+import { StylesApiProps } from "props/styles";
+import React from "react";
+import { getLoadingState, getChildLayout } from "../../../utils/dash3";
+
+interface Props extends __PopoverProps, StylesApiProps {
+    variant?: string;
+    /** Menu content */
+    children?: React.ReactNode;
+    /** Controlled menu opened state */
+    opened?: boolean;
+    /** Uncontrolled menu initial opened state */
+    defaultOpened?: boolean;
+    /** Determines whether dropdown should trap focus of keyboard events */
+    trapFocus?: boolean;
+    /** Determines whether Menu should be closed when item is clicked */
+    closeOnItemClick?: boolean;
+    /** Determines whether arrow key presses should loop though items (first to last and last to first) */
+    loop?: boolean;
+    /** Determines whether dropdown should be closed when Escape key is pressed */
+    closeOnEscape?: boolean;
+    /** Event which should open menu */
+    trigger?: "click" | "hover" | "click-hover";
+    /** Open delay in ms, applicable only to trigger="hover" variant */
+    openDelay?: number;
+    /** Close delay in ms, applicable only to trigger="hover" variant */
+    closeDelay?: number;
+    /** Determines whether dropdown should be closed on outside clicks */
+    closeOnClickOutside?: boolean;
+    /** Events that trigger outside clicks */
+    clickOutsideEvents?: string[];
+    /** Set the `tabindex` on all menu items. Defaults to -1 */
+    menuItemTabIndex?: -1 | 0;
+    /** Unique ID to identify this component in Dash callbacks. */
+    id?: string;
+    /** Update props to trigger callbacks. */
+    setProps: (props: Record<string, any>) => void;
+}
+
+/** Sub Menu */
+const MenuSub = (props: Props) => {
```

Comment:
Should it be called `SubMenu` instead?

Reply:
Not a real strong opinion but I don't think it would be too big a stretch to let `Menu.Sub` translate to `SubMenu`. Some namespaced components make sense in the original order, some not so much. I feel like just pointing this out in the docstring would suffice, like "`SubMenu` is a Mantine `Menu.Sub` component."

Reply:
Yeah, I kinda like that better too.